Fraction_class.py

```diff
@@ -0,0 +1,87 @@
+#def a Fraction class, which has two attributes: numerator and denomitor, and three methods: __str__,__add__,__eq__
+#understand the differences between shallow equation and deep equation
+def gcd_euclid_recursion(m,n):
+    m=abs(m)
+    n=abs(n)
+    if m==n:
+        return m
+    elif m<n:
+        m,n=n,m
+    if m%n==0:
+        return n
+    else:
+        temp=m%n
+        m=n
+        n=temp
+        euc=gcd_euclid_recursion(m,n)

```

Comment:
should be written as `return gcd_euclid_recursion(m,n)`, tail recursive. 👍

Reply:
what does tail recursive mean?